akka-http/client/src/main/scala/endpoints/akkahttp/client/Urls.scala

```diff
@@ -0,0 +1,87 @@
+package endpoints.akkahttp.client
+
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets.UTF_8
+
+import endpoints.Tupler
+import endpoints.algebra
+
+/**
+  * [[algebra.Urls]] interpreter that builds URLs.
+  */
+trait Urls extends algebra.Urls {
+
+  val utf8Name = UTF_8.name()
+
+  trait QueryString[A] {
+    def encodeQueryString(a: A): Option[String]
+  }
+
+  def combineQueryStrings[A, B](first: QueryString[A], second: QueryString[B])(implicit tupler: Tupler[A, B]): QueryString[tupler.Out] =
+    (ab: tupler.Out) => {
+      val (a, b) = tupler.unapply(ab)
+
+      (first.encodeQueryString(a), second.encodeQueryString(b)) match {
+        case (Some(left), Some(right)) => Some(s"$left&$right")
+        case (Some(left), None) => Some(left)
+        case (None, Some(right)) => Some(right)
+        case (None, None) => None
+      }
+    }
+
+  def qs[A](name: String)(implicit value: QueryStringParam[A]): QueryString[A] =
+    a => Some(s"$name=${value.apply(a)}")
+
+  def optQs[A](name: String)(implicit value: QueryStringParam[A]): QueryString[Option[A]] = {
+    case Some(a) => qs[A](name).encodeQueryString(a)
+    case None => None
+  }
+
+  type QueryStringParam[A] = A => String
+
+  implicit lazy val stringQueryString: QueryStringParam[String] = s => URLEncoder.encode(s, utf8Name)
```

Comment:
There are many useful URI and Query types/methods in https://github.com/akka/akka-http/blob/master/akka-http-core/src/main/scala/akka/http/scaladsl/model/Uri.scala#L601